### PR TITLE
Setup comprehensive test codebase

### PR DESCRIPTION
**Pull request by ItsBruno**
## Summary

This PR refactors the existing java tests suites to Kotlin (but does not remove them yet for safety) and adds new test suites for non covered classes. The idea is to catch potential bugs in the Java -> Kotlin refactor of the PoV library. 

## Changes

- Added copilot PR comment suggestion from feature/raise-kotlin-version pull request
- Refactored existing test suites to Kotlin and added additional cases (the existing ones are not removed for extra safety during the Java -> Kotlin rewrite, given that the refactored ones should cover the existing ones they can be removed after the refactor)
- Added new test suites for non covered classes

### Type

- [ ] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [X] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

### Description

Test names don't follow the usual backtick nomenclature as I wanted to keep the codebase consistent with existing tests in the repository. If this is an issue I can change it.

The existing java test cases are not removed for extra safety during the Java -> Kotlin rewrite as they currently test working production code. Given that the refactored ones should cover the existing java ones the java tests can be removed after the refactor.

Tests `testEnqueueWhenCanceledCallsOnError` and `testEnqueueWithExecutorWhenCanceledCallsOnError ` in `UpdaterCallTest` fail because there is no return statement in the cancel if blocks. Currently they are commented out. This behaviour should be checked and fixed during the Java -> Kotlin refactor and the tests uncommented.

Test `checkEqualsAndHashCodeContract` fails when checking the hash code for optional updates due to an exception, ideally hash code should not throw exceptions. During the Java -> Kotlin refactor it should be checked whether this bug is of real concern in the actual implementation and context of usage if the hash code will never be checked for optional updates. If the bug is addressed the test should be updated.

## Checklist

- [X] I have performed a self-review of my own code.
- [X] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

### CODE REVIEW

File: prince-of-versions/src/test/java/co/infinum/princeofversions/ApplicationConfigurationImplTest.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import android.content.pm.PackageInfo
+import android.content.pm.PackageManager
+import android.os.Build
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+
+class ApplicationConfigurationImplTest {
```
Comment (KCeh) [inline prince-of-versions/src/test/java/co/infinum/princeofversions/ApplicationConfigurationImplTest.kt:13]: I hope you use AI to generate this tests 😅

**Review by KCeh — COMMENTED** [review]
> The existing java test cases are not removed for extra safety during the Java -> Kotlin rewrite as they currently test working production code. Given that the refactored ones should cover the existing java ones the java tests can be removed after the refactor.

👍 

It is kind of hard to review all of the code properly, since there is quite a lot of it.
And this test definitely look like AI generated. 
But as long all is ok with them (and they all pass) 👍 
Also, really interesting approach to development. If you learned something interesting write it down and you can pass it to the reset of developers later on

File: prince-of-versions/src/test/java/co/infinum/princeofversions/CheckResultTestRefactored.kt [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+        // Test hashCode contract
+        // For optional updates, hashcode should be consistent
+        assertThat(optionalResult1.hashCode()).isEqualTo(optionalResult2.hashCode())
+
+        // For mandatory updates, hashCode throws an exception - this is a bug in the class, but the test should document it
+        // TODO - check whether this bug is of real concern in the actual implementation and context of usage - if so fix it
+        assertThatThrownBy { mandatoryResult1.hashCode() }
+            .isInstanceOf(UnsupportedOperationException::class.java)
+            .hasMessage("There is no optional update available.")
+    }
```
[the remark this answers is not in the document]
Comment (ItsBruno) [reply prince-of-versions/src/test/java/co/infinum/princeofversions/CheckResultTestRefactored.kt:288]: You are right, it does have a `hashCode` implementation. The issue is caused by the `getNotificationType()` function used in hash code generation. That method can throw an exception in specific cases causing `hashCode` to crash the app. This is not valid behaviour for the `hashCode` function.

[the remark this answers is not in the document]
Comment (KCeh) [reply prince-of-versions/src/test/java/co/infinum/princeofversions/CheckResultTestRefactored.kt:288]: Can we fix it?
